src/states/elder/mod.rs

```diff
@@ -381,34 +380,31 @@ impl Elder {
         }
     }
 
-    fn finalise_prefix_change(&mut self) -> Result<(), RoutingError> {
-        // Clear any relocation overrides
-        self.next_relocation_dst = None;
-        self.next_relocation_interval = None;
-
+    fn reset_parsec_with<F: Fn(&mut Chain) -> Result<ParsecResetData, RoutingError>>(
+        &mut self,
+        get_reset_data: F,
```

Comment:
Why not pass the reset data directly instead of through the closure?

Reply:
Good point. I think this piece of code started out as something that had to be called after something else, but then I moved things around and never changed this approach.